pep-9999.rst

```diff
@@ -0,0 +1,1037 @@
+PEP: 9999
+Title: Graceful cooperation between external and Python package managers
+Author: Geoffrey Thomas <geofft@ldpreload.com>, Matthias Klose <doko@ubuntu.com>, Filipe Laíns <lains@riseup.net>, Donald Stufft <donald@python.org>, Tzu-Ping Chung <uranusjr@gmail.com>, Stefano Rivera <stefanor@debian.org>, Elana Hashman <ehashman@debian.org>, Pradyun Gedam <mail@pradyunsg.me>
+Discussions-To: TODO discourse
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 2021-05-18
+Post-History:
+
+Abstract
+========
+
+This PEP proposes a mechanism to indicate that a certain
+Python interpreter's package installation is managed by
+external means (such as an OS package manager) and that
+Python-specific package management tools like pip should, by
+default, neither install nor remove packages in an
+interpreter-wide context, guiding the end user towards using
+a virtual environment.
+
+It also standardizes an interpretation of the ``sysconfig``
+schemes so that, if a Python-specific package manager is
+specifically instructed to install a package in an
+interpreter-wide context, it can do so in a manner that
+avoids conflicting with the external package manager and
+reduces the risk of breaking software shipped by the
+external package manager.
+
+Motivation
+==========
+
+Thanks to Python's immense popularity, software
+distributions (by which we mean Linux and other OS
+distributions as well as overlay distributions like Homebrew
+and MacPorts) generally ship Python for two purposes: as a
+software package to be used in its own right by end users,
+and as an implementation language for other software in the
+distribution.
+
+For example, many distributions ship a ``/usr/bin/python3``
+command which is both the ``python3`` command available to
+end users as well as the ``#!/usr/bin/python3`` shebang for
+Python-language software. Because there are no official
+binary releases of Python for Linux/UNIX, almost all Python
+end users on these OSes use the Python interpreter built and
+shipped with their distribution.
```

Comment:
```suggestion
binary releases of Python for Linux/UNIX, almost all Python
end users on these operating systems use the Python interpreter built and
shipped with their distribution in some way, even if they compile their
own Python binaries.
```

Reply:
Hm I think this changes the focus of the statement away from "most people do have a dual-use Python" to "most people use the distro Python interpreter in at least one of the two ways above," but if they only use it in one way then the problem kind of goes away. What's the change you're trying to convey here?

Reply:
A lot of people will claim it's bad practice to have a dual-use Python, when in reality they still use the system Python in a non-trivial way, whether or not they realize it.

Reply:
How about s/almost all/most/, and then a new paragraph after the mention of `pip install --user`, something like

> Even for Python end users who avoid using the system-provided Python interpreter, such as by compiling their own from source, packages installed by either interpreter via `sudo pip install` or `pip install --user` will be visible on the `sys.path` of both interpreters. (As discussed later in this document, some distros configure their interpreter to use a non-default location for `sudo pip install` to address this use case, but no known distro customizes the `pip install --user` location / the user site-packages directory.)